### ISSUE TITLE
💡 Power Status Notification threshold

### ISSUE THREAD

**Issue by xeruf** [manjaro-sway/manjaro-sway#480]
### Summary

Not sure which application manages it, but the power status notifications get a bit excessive as my framework, when fully charged, will stop charging at 100% and then start again at 97%, triggering notifications every few minutes.

### Drawbacks

Genuine notifications might be missed.

### Implementation idea

Maybe a threshold to disable notifications once topped up until the battery drops below a certain percentage like 90% for once?

### Will you be able to contribute a PR for this?

- [ ] Yes I will be able to get that feature a *lot* quicker.

**Comment by boredland** [comment]
It is using poweralertd (https://sr.ht/~kennylevinsen/poweralertd/). Perhaps there is some flag for that, but honestly on my machines I never observed too many notifications...

**Comment by boredland** [comment]
> when fully charged, will stop charging at 100% and then start again at 97%

Am I understanding that correctly: without you unplugging, it charges to 100% and then discharges to 97%? That sounds wrong. Perhaps I don't fully understand the scenario yet.

**Comment by xeruf** [comment]
Yes, this is exactly what seems to happen, I guess for battery longevity. Not too surprising imo, but maybe I am misinterpreting what is going on.

**Comment by boredland** [comment]
I think this is a hardware specific problem. Poweralertd does what it is supposed to do here.

**Comment by xeruf** [comment]
it is not a problem per se, it is theoretically correct, just a nuisance that might be avoided
for now I usually disable popup notifications altogether, as it also keeps popping up if I use my laptop with a charger that hovers around the wattage that my laptop draws, switching between charging and decharging

**Comment by boredland** [comment]
I today had this the first time ever on my thinkpad x1 (g6)

**Comment by boredland** [comment]
there is an ongoing discussion about this: https://lists.sr.ht/~kennylevinsen/poweralertd-devel/patches/43895

I think all proposals, including that patch, could be good enough.

**Comment by xeruf** [comment]
Still waiting on that fix, got even more annoying as I am now on a Framework 16 where I limited max battery to 60% for stationery use, so the notifications pop around that battery level, so those fixes do not work properly either.

**Comment by boredland** [comment]
Ya, newly framework 13 user here and same issue. As I understand those upstream proposals, this is also a framework issue, as it keeps sending those "useless" state changes, even if the battery isn't being charged.

**Comment by xeruf** [comment]
I now have an even trickier one:
I use a charger with a wattage just enough for the laptop, meaning the charge can hover about whatever percentage it is when plugging in, with regular charge/discharge notifications.
Is it possible to simply define a threshold filter, that only lets notifications through below 30%?

**Comment by xeruf** [comment]
What is the middleware responsible for displaying notifications? Maybe it can be configured to have a filter.

**Comment by boredland** [comment]
I think I get what you mean. Guess we could start mako with a configured filter string that is read from a file? According to `man 5 mako` this could work.

**Comment by xeruf** [comment]
Ah now that's a lot nicer!
![image](https://github.com/user-attachments/assets/595a7773-c770-470a-8ca9-3f472056a6c8)
Fund the [Example configuration](https://github.com/emersion/mako/wiki/Example-configuration) linked [here](https://www.lorenzobettini.it/2023/11/hyprland-and-notifications-with-mako/).

**Comment by xeruf** [comment]
btw any particular reason of choosing mako above something like dunst?

**Comment by boredland** [comment]
> I think the default mako configuration should not be split between CLI args and /usr/share/sway/templates/mako, instead they should be unified in a skel file populating the users home directory

The primary goal to have this in the sway config is, that we can do multiple themes. To have a clear path to opt out would still be nice..

**Comment by xeruf** [comment]
What do you mean by multiple themes?

**Comment by boredland** [comment]
Like with the `manjaro-sway-themes` rofile application we have.

**Comment by xeruf** [comment]
anyways, my DND still showing high prio works, cause battery low notifications are sent on high prio - only thing left is to filter that even outside DND and post it here for completeness :)

**Comment by robin-thoene** [comment]
> Ya, newly framework 13 user here and same issue. As I understand those upstream proposals, this is also a framework issue, as it keeps sending those "useless" state changes, even if the battery isn't being charged.

I can confirm this being an issue on my framework 13 as well. Although it is not consistent. Some days I get more notifications, some days none.

**Comment by Trust2k** [comment]
Я отключил уведомления батареи, вот так:

> [app-name="poweralertd"]
invisible=1
to ~/.config/mako/config and restarted mako.

https://forum.manjaro.org/t/disable-power-status-notifications/125414/3
